pydantic/_internal/_generate_schema.py

```diff
@@ -2425,12 +2392,108 @@ def _common_field(
     }
 
 
+class CollectedInvalid(Exception):
+    pass
+
+
+_NOT_INLINABLE_META_KEYS: set[str] = {
+    *CoreMetadata.__annotations__.keys(),
+    _discriminated_union.CORE_SCHEMA_METADATA_DISCRIMINATOR_PLACEHOLDER_KEY,
+    TAGGED_UNION_TAG_KEY,
+}
+
+
 class _Definitions:
     """Keeps track of references and definitions."""
 
     def __init__(self) -> None:
-        self.seen: set[str] = set()
-        self.definitions: dict[str, core_schema.CoreSchema] = {}
+        self.used_deferred_discriminators = False
+        self._recursively_seen: set[str] = set()
+        self._definitions: dict[str, CoreSchema] = {}
+        # Def schema instances of another model, not to be exposed via get_def
+        self._unpacked_definitions: dict[str, CoreSchema] = {}
+
+    def get_def(self, ref: str) -> CoreSchema | None:
+        return self._definitions.get(ref)
```

Comment:
Is this adding much value?

Reply:
So adding `__getitem__` instead?

Reply:
I wouldnt make it public to not cause a mayhem of things mutating the registry

Reply:
I guess I was just thinking that `self.defs._definitions.get(ref)` wasn't so bad, but I get that's not ideal (`self` here being a `GenerateSchema` instance).

Maybe a compromise, naming it `get_from_ref`? Seems like `self.defs.get_from_ref(some_ref)` sheds more light on the functionality? or `get_def_from_ref`, but that seems verbose given that we're calling it on `defs`...

Reply:
Done, renamed to `get_from_ref`